src/overlays/actors/ovl_Arrow_Ice/z_arrow_ice.c

```diff
@@ -21,21 +33,214 @@ const ActorInit Arrow_Ice_InitVars = {
+    this->actor.shape.rot = arrow->actor.shape.rot;
+
+    func_8002F974(&this->actor, NA_SE_PL_ARROW_CHARGE_ICE);
+
+    // If arrow's attached is null, Link has fired the arrow
+    if (arrow->actor.attachedA == NULL) {
+        this->unkPos = this->actor.posRot.pos;
+        this->radius = 10;
+        ArrowIce_SetupAction(this, ArrowIce_Fly);
+        this->alpha = 0xFF;
+    }
+}
+
+void func_80867E8C(Vec3f* unkPos, Vec3f* icePos, f32 scale) {
+    unkPos->x += ((icePos->x - unkPos->x) * scale);
+    unkPos->y += ((icePos->y - unkPos->y) * scale);
+    unkPos->z += ((icePos->z - unkPos->z) * scale);
+}
+
+void ArrowIce_Hit(ArrowIce* this, GlobalContext* globalCtx) {
+    f32 scale;
+    f32 offset;
+    u16 timer;
+
+    if (this->actor.unk_F0 < 50.0f) {
+        scale = 10.0f;
+    } else {
+        if (950.0f < this->actor.unk_F0) {
+            scale = 310.0f;
+        } else {
+            scale = this->actor.unk_F0;
+            scale = ((scale - 50.0f) * 0.33333334f) + 10.0f;
+        }
+    }
+
+    timer = this->timer;
+    if (timer != 0) {
+        this->timer -= 1;
+
+        if (this->timer >= 8) {
+            offset = ((this->timer - 8) * 0.041666668f);
+            offset = SQ(offset);
+            this->radius = (((1.0f - offset) * scale) + 10.0f);
+            this->unk_160 += ((2.0f - this->unk_160) * 0.1f);
+            if (this->timer < 16) {
+                if (1) {}
+                this->alpha = ((this->timer * 0x23) - 0x118);
+            }
+        }
+    }
+
+    if (this->timer >= 9) {
+        if (this->unk_164 < 1.0f) {
+            this->unk_164 += 0.25f;
+        }
+    } else {
+        if (this->unk_164 > 0.0f) {
+            this->unk_164 -= 0.125f;
+        }
+    }
+
+    if (this->timer < 8) {
+        this->alpha = 0;
+    }
+
+    if (this->timer == 0) {
+        this->timer = 255;
+        Actor_Kill(&this->actor);
+    }
+}
+
+void ArrowIce_Fly(ArrowIce* this, GlobalContext* globalCtx) {
+    EnArrow* arrow;
+    f32 distanceScaled;
+    s32 pad;
+
+    arrow = this->actor.attachedA;
+    if ((arrow == NULL) || (arrow->actor.update == NULL)) {
+        Actor_Kill(&this->actor);
+        return;
+    }
+    // copy position and rotation from the attached arrow
+    this->actor.posRot.pos = arrow->actor.posRot.pos;
+    this->actor.shape.rot = arrow->actor.shape.rot;
+    distanceScaled = Math_Vec3f_DistXYZ(&this->unkPos, &this->actor.posRot.pos) * 0.041666668f;
+    this->unk_160 = distanceScaled;
+    if (distanceScaled < 1.0f) {
+        this->unk_160 = 1.0f;
+    }
+    func_80867E8C(&this->unkPos, &this->actor.posRot.pos, 0.05f);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arrow_Ice/ArrowIce_Destroy.s")
+    if (arrow->hitWall & 1) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_IT_EXPLOSION_ICE);
+        ArrowIce_SetupAction(this, &ArrowIce_Hit);
+        this->timer = 32;
+        this->alpha = 0xFF;
+    } else if (arrow->timer < 0x22) {
```

Comment:
I think decimal would be more appropriate here since it's a timer value.
```suggestion
    } else if (arrow->timer < 34) {
```